plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/RewriteStringComparison.java

```diff
@@ -70,19 +70,48 @@ public Optional<ParameterizedExpression> rewrite(Call expression, Captures captu
         Variable firstArgument = captures.get(FIRST_ARGUMENT);
         Variable secondArgument = captures.get(SECOND_ARGUMENT);
 
-        if (isClob(firstArgument, context) || isClob(secondArgument, context)) {
-            return Optional.empty();
+        boolean[] clobInfo = {false, false};
+        if (isClob(firstArgument, context)) {
+            clobInfo[0] = true;
+        }
+        if (isClob(secondArgument, context)) {
+            clobInfo[1] = true;
         }
         return context.defaultRewrite(firstArgument).flatMap(first ->
                 context.defaultRewrite(secondArgument).map(second ->
                         new ParameterizedExpression(
-                                "(%s) %s (%s)".formatted(first.expression(), comparison.getOperator(), second.expression()),
+                                buildComparison(first.expression(), comparison.getOperator(), second.expression(), clobInfo),
                                 ImmutableList.<QueryParameter>builder()
                                         .addAll(first.parameters())
                                         .addAll(second.parameters())
                                         .build())));
     }
 
+    private static String buildComparison(
+            String firstExpression,
+            String operator,
+            String secondExpression,
+            boolean[] clobInfo)
+    {
+        if (clobInfo[0] || clobInfo[1]) {
+            String firstArgument = firstExpression;
+            String secondArgument = secondExpression;
+            if (!clobInfo[0]) {
+                firstArgument = toNClob(firstExpression);
+            }
+            if (!clobInfo[1]) {
+                secondArgument = toNClob(secondExpression);
+            }
+            return "DBMS_LOB.compare(%s, %s) %s 0".formatted(firstArgument, secondArgument, operator);
```

Comment:
Is the DBMS_LOB is something which is provided by default on Oracle or is it some sort of a package which needs to be installed ? How would it work with collation ? What if the underlying bytes in CLOB has some invalid string pattern - in this case won't Trino and Oracle differ ?

Reply:
removed this change. nclob comparison is out of scope of this PR.